File: python/剑指offer/面试题57 - II. 和为s的连续正数序列.py
from typing import List
class Solution:
    def findContinuousSequence(self, target: int) -> List[List[int]]:
        import itertools
        result = []
        for i in itertools.count(1):
            j = i+1
            sum = i+j
            if sum > target:
                break
            while sum < target:
                j = j+ 1
                sum += j
            if sum == target:
                result.append(list(range(i,j+1)))
        return result

s = Solution()
target = 15
print(s.findContinuousSequence(target))

File: go/剑指offer/面试题57 - II. 和为s的连续正数序列.go
package main

func findContinuousSequence(target int) [][]int {
	i, j := 1, 1
	var res [][]int
	sum := 0
	for i < target/2 {
		if sum < target {
			sum += j
			j += 1
		} else if sum > target {
			sum -= i
			i += 1
		} else {
			var tmp []int
			for z := i; z < j; z++ {
				tmp = append(tmp, z)
			}
			res = append(res, tmp)
			sum -= i
			i += 1
		}
	}
	return res
}
